woocommerce/woocommerce-android
# Issue/6339 react to beta switch

Closes: #6339


### Description
This PR adds a way to listen to preferences changes using the AppPrefsWrapper, which allows reacting to the changes and refreshing the UI, the goal here is to fix the issue of not refreshing the menu buttons when the coupon toggle is refreshed.

### Testing instructions
1. Make sure "Don't keep activities" is disabled (this was necessary to reproduce the original issue)
2. Go to Settings > Beta Features
3. Turn on Coupon Management
4. Go back to the hub menu screen
5. Confirm the coupons button is visible correctly.
